### ISSUE TITLE
resolution parameter for incoming observations

### ISSUE THREAD

**Issue by kvas7andy** [minosworld/minos#20]
Hi, everyone!

I found useful to retrieve necessary information form sensors, which I can get from sensors.
For example, to get rgb image observation I do `observation['observation']['sensors']['color']['data']`
But resolution of output data (arrays) is (128, 128). After trying to change resolution of observations rendering  from simulator in _sensors.yml_  as well as via _env_config_ files, I didn't reach any solution.

Could you please tell more about resolution changing (for observations, not for human rendering)?

**Comment by angelxuanchang** [comment]
Can you clarify what you mean by resolution changing for the observations vs environment rendering?  The two are the same and are controlled by external parameters in the command line arguments (such as `--width` and `--height` see https://github.com/minosworld/minos/blob/master/config/sim_args.py#L19)

If you look carefully at `sensors.yml`, you will notice that there is a `resize: true`.  That allows the resolution of the sensor to be overridden and resized by the external command line parameters.

**Comment by angelxuanchang** [comment]
Yes, you are right.  Please let us know if you see that this is not the case.  You can test out the `--width` and `--height` options with the gym demo:
`python3 demo.py --env_config objectgoal_suncg_mf --width 256 --height 128`

**Comment by kvas7andy** [comment]
Thank you for your patience and attention.

It required me some time to check if the problem still exists.
So, for equal  width and height rendering works fine as well as observation output
For unequal  width and height, rendering works fine, but observation output for sensors is given in reversed order (width replaced with height) and image is distorted.
I think, the problem is similar to previous problem which was solved by @msavva, here https://github.com/minosworld/minos/issues/9

So the problem still exists.

**Comment by kvas7andy** [comment]
@angelxuanchang  
I couldn't find my own workaround, so I would like to ask your help in this problem, stated above (my previous post)

**Comment by msavva** [comment]
@kvas7andy I am unable to reproduce the issue you observe.  Running `python3 demo.py --env_config pointgoal_suncg_mf --width 256 --height 128` in `minos/gym` produces output as below:

![screenshot 2018-03-13 19 51 14](https://user-images.githubusercontent.com/455005/37380920-f491d79e-26f7-11e8-9e61-0c26b9ccc692.png)

Can you confirm that you have cloned a clean copy of the repository and compiled the minos server from scratch?  If you have multiple minos repositories, note that you may need to `pip uninstall minos` and `pip install -e .` in the code repository you want to use.

**Comment by msavva** [comment]
@kvas7andy I believe I understand why you observe the issue you reported.  You must be pulling out `img = observation['observation']['sensors']['color']['data']` and saving the numpy array as an image directly.  In order for this image to look correct you will need to reshape such that the height is the first dimension using `img = img.reshape((img.shape[1], img.shape[0], img.shape[2]))`.

**Comment by msavva** [comment]
@kvas7andy Are you able to use the sensor data as you would expect? If yes, shall we close this issue?  I will summarize some of the details of this discussion in a FAQ document so that others can also refer to it.

**Comment by kvas7andy** [comment]
@msavva it would be great to have Friday to make tests!

**Comment by msavva** [comment]
@kvas7andy the "crossboard" pattern is usually how transparency is visualized by image viewers (i.e. alpha values <1 in RGBA images).  Can you copy an output image you get so we can verify whether this is the expected behavior?

**Comment by msavva** [comment]
@kvas7andy are there any remaining issues with the returned observation images?

**Comment by kvas7andy** [comment]
`plt.imshow()` and `scipy.misc.imsave()` have "crossboard" outputs just in my own viewer. After uploading images in github comment (drag-n-drop) crossboards vanished. Thank you for clarification. 

![5_3_img](https://user-images.githubusercontent.com/8929593/38682937-f9b138fc-3e74-11e8-9f52-65f528884f28.png)
![1_2_img](https://user-images.githubusercontent.com/8929593/38682938-f9cc5f10-3e74-11e8-8526-9c0a85878155.png)
